static/js/components/VideoPlayer.js

```diff
@@ -1,37 +1,216 @@
 // @flow
+/* global videojs: true */
 import React from 'react';
 
-import HTML5VideoPlayer from "./HTML5VideoPlayer";
-import USwitchPlayer from "./USwitchPlayer";
+import { makeVideoSubtitleUrl } from "../lib/urls";
+import { getHLSEncodedUrl, videojs } from "../lib/video";
+import type { Video, VideoSubtitle } from "../flow/videoTypes";
+import { FULLSCREEN_API } from "../util/fullscreen_api";
+import { CANVASES } from "../constants";
 
-import { makeEmbedUrl } from "../lib/urls";
+const makeConfigForVideo = (video: Video): Object => ({
+  autoplay: true,
+  controls: true,
+  fluid: false,
+  playsinline: true,
+  html5: {
+    nativeTextTracks: false
+  },
+  sources: [{
+    src: getHLSEncodedUrl(video),
+    type: 'application/x-mpegURL',
+  }]
+});
 
-import type { Video } from '../flow/videoTypes';
+const drawCanvasImage = function(canvas, videoNode, shiftX, shiftY) {
+  const x = shiftX ? Math.floor(videoNode.videoWidth / 2) : 0;
+  const y = shiftY ? Math.floor(videoNode.videoHeight / 2) : 0;
+  const context = canvas.getContext('2d');
+  context.drawImage(
+    videoNode,
+    x, y, Math.floor(videoNode.videoWidth / 2), Math.floor(videoNode.videoHeight / 2),
+    0, 0, canvas.width, canvas.height
+  );
+  setTimeout(drawCanvasImage, 20, canvas, videoNode, shiftX, shiftY);
```

Comment:
[This](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) says that `setTimeout` can't be used this way in IE9 and below. Do you know if we support any of those browsers for this app?

Reply:
@pdpinch does OVS need to support IE9 and below?

Reply:
@noisecapella that links says a polyfill can make this work in IE, maybe `babel-polyfill` does that automatically? - https://github.com/zloirock/core-js#settimeout--setinterval

Reply:
Yes I think you're right